src/mkfs.fat.c

```diff
@@ -593,7 +594,7 @@ static void establish_params(struct device_info *info)
     }
 
     if (!hidden_sectors_by_user && info->geom_start >= 0)
-	hidden_sectors = htole32(info->geom_start);
+        hidden_sectors = htole32(info->geom_start) + part_offset;
```

Comment:
And this too... `hidden_sectors` is in sectors unit. Therefore `part_offset` is really in sectors?